### DISCUSSION TITLE
Customizable badge colors

### DISCUSSION THREAD

**Discussion by ZachNagengast** [SwiftPackageIndex/SwiftPackageIndex-Server#2886] [Ideas]
It would be nice to be able to customize the color of the badge through the api. As much as I love the swift red - red is also the color used when builds/tests are failing, since these badges are usually clustered together it would be great if all the colors could match when the repo is in a "passing" state.
Example:
<img width="345" alt="image" src="https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/assets/1981179/c264d169-27ba-49d0-8b78-e39a6da0957c">

It may be as simple as adding another param here to the BadgeQuery for `color`

https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/blob/93e4e4ebc772d5b940c47a5a9c52d62880c35ab5/Sources/App/Controllers/API/API%2BPackageController.swift#L35

Currently, it appears to just be a hardcoded hex as long as the badge has valid values:

https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/blob/93e4e4ebc772d5b940c47a5a9c52d62880c35ab5/Sources/App/Core/Extensions/Badge.swift#L55

**Comment by daveverwer** [comment]
Thanks for the suggestion @ZachNagengast! I hadn't seen it as a "failing" color.

The only thing that makes me hesitate is that our badges are intended to indicate compatibility rather than to be any kind of CI, and we deliberately didn't use green for that reason. The difference is subtle, but allowing a customised colour might take it one step closer to being seen as CI.

The only other thing I'd want to keep in mind is to keep the UI for copying the Markown nice and simple. It would be possible to add a colour field there, but I'm not sure I want to make that more complex, and without something along those lines it would be hard to communicate that customisation was available.

I'm not dead against it, but I'm also not sure I'm convinced it's a good idea yet 😂

**Reply by ZachNagengast** [reply]
Understandable, I'd definitely consider it an optional customization, for folks that want to override the default behavior. Anyone seeking out that customization would likely be familiar with shields.io api syntax with the various optional params like `color` and would be able to use shields.io directly if they want. 

e.g. space grey:
![](https://img.shields.io/endpoint?url=https%3A%2F%2Fswiftpackageindex.com%2Fapi%2Fpackages%2FSwiftPackageIndex%2FSwiftPackageIndex-Server%2Fbadge%3Ftype%3Dswift-versions&logoColor=000000&color=717378)

Either way just an idea, thanks for the response!
